crates/uv-cli/src/lib.rs

```diff
@@ -4797,6 +4797,22 @@ pub struct PublishArgs {
         value_parser = parse_insecure_host,
     )]
     pub allow_insecure_host: Option<Vec<Maybe<TrustedHost>>>,
+
+    /// Don't upload files that already exists on the index. The value is the index URL.
+    ///
+    /// This option allows retrying publishing that failed after only some, but not all files have
+    /// been uploaded, and handles error due to parallel uploads of the same file.
+    ///
+    /// Before uploading, the index is checked. If the exact same file already exists in the index,
+    /// the file will not be uploaded. If an error occurred during the upload, the index is checked
+    /// again, if the file may have been added in the meantime.
+    ///
+    /// The exact behavior varies from index to index. On PyPI, uploading the same file succeeds
```

Comment:
I'd say something like "The exact behavior will vary based on implementation details of the index. When uploading to PyPI, ..."